onnx/defs/tensor/defs.cc

```diff
@@ -2896,14 +2888,17 @@ ONNX_OPERATOR_SET_SCHEMA(
           } else {
             // 'axis' is provided.
             int axis = static_cast<int>(axisAttr->i());
-            const TensorShapeProto& input_shape =
+            if(!xTensorProto->tensor_type().has_shape()) {
```

Comment:
should this part be updated to check with `(nullptr == xTensorProto) || !xTensorProto->tensor_type().has_shape()`?

Reply:
it's not required in this case. This will be checked as part of the type inference.
see line 2849 ->propagateElemTypeFromInputToOutput(ctx, 0, 0);